File: tools/Word Document Editor/scripts/list_changes.py
#!/usr/bin/env python3
"""
List all track changes in a Word document.

Usage:
    python3 list_changes.py document.docx
"""

import sys
from pathlib import Path
from docx_utils import list_track_changes


def main():
    if len(sys.argv) < 2:
        print("Usage: python3 list_changes.py document.docx", file=sys.stderr)
        sys.exit(1)
    
    docx_path = sys.argv[1]
    
    try:
        changes = list_track_changes(docx_path)
        
        # Print results
        if not changes:
            print("✓ No track changes found in document.")
            return
        
        print(f"\nFound {len(changes)} track change(s):\n")
        print("=" * 80)
        
        for change in changes:
            print(f"\nChange ID: {change['id']}")
            print(f"Type: {change['type']}")
            print(f"Author: {change['author']}")
            if change['date']:
                print(f"Date: {change['date']}")
            print(f"Text: {repr(change['text'][:100])}{'...' if len(change['text']) > 100 else ''}")
            print(f"Status: {change['status']}")
            print("-" * 80)
    
    except FileNotFoundError as e:
        print(f"❌ Error: {e}", file=sys.stderr)
        sys.exit(1)
    except ValueError as e:
        print(f"❌ Error: {e}", file=sys.stderr)
        sys.exit(1)
    except Exception as e:
        print(f"❌ Error: {type(e).__name__}: {str(e)}", file=sys.stderr)
        import traceback
        traceback.print_exc()
        sys.exit(1)




if __name__ == '__main__':
    main()

File: tools/Word Document Editor/scripts/list-changes.js
#!/usr/bin/env node
/**
 * List all track changes in a Word document.
 *
 * Usage:
 *     node list-changes.js document.docx
 */

import { listTrackChanges } from './docx-utils.js';
import { existsSync } from 'fs';

// CLI
function showHelp() {
  console.log(`
List all track changes in a Word document

Usage:
  node list-changes.js <document>

Arguments:
  document    Path to Word document (.docx)

Options:
  --help, -h  Show this help message
`);
  process.exit(0);
}

const args = process.argv.slice(2);

if (args.length === 0 || args.includes('--help') || args.includes('-h')) {
  showHelp();
}

const docxPath = args[0];

if (!existsSync(docxPath)) {
  console.error(`Error: File not found: ${docxPath}`);
  process.exit(1);
}

try {
  const changes = await listTrackChanges(docxPath);

  if (changes.length === 0) {
    console.log('No track changes found in document.');
    process.exit(0);
  }

  console.log(`\nFound ${changes.length} track change(s):\n`);
  console.log('='.repeat(80));

  for (const change of changes) {
    console.log(`\nChange ID: ${change.id}`);
    console.log(`Type: ${change.type}`);
    console.log(`Author: ${change.author}`);
    if (change.date) {
      console.log(`Date: ${change.date}`);
    }
    const textPreview = change.text.length > 100 ? change.text.slice(0, 100) + '...' : change.text;
    console.log(`Text: "${textPreview}"`);
    console.log(`Status: ${change.status}`);
    console.log('-'.repeat(80));
  }
} catch (e) {
  console.error(`Error: ${e.message}`);
  process.exit(1);
}

